recipes/geographiclib/all/conanfile.py

```diff
@@ -23,9 +24,24 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    @property
+    def _min_compiler_version_default_cxx11(self):
+        # Minimum compiler version having c++ standard >= 11
+        return {
+            "gcc": 6,
+            "clang": 6,
+            "Visual Studio": 14,  # guess
```

Comment:
```suggestion
            "gcc": 4.9,
            "clang": 6,
            "Visual Studio": 14,  # guess
```

I'm able to build with gcc 4.9 and 5

Reply:
"gcc": 4.9 -> "gcc": 6.1
The default mode is C++98 for GCC versions prior to 6.1, and C++14 for GCC 6.1 and above.

Reply:
It supports C++11 even if it's not the default.

Reply:
it doesn't matter, the important thing is that it has only c++03 default mode.

And  c++11 support in gcc 4.9 is quite buggy. And anyway, what's the point of using  almost EOL compiler?

1. Use compiler.cppstd if set (  On CCI cppstd is not set anywhere)
2. Use C++ standard compiler _default mode_, if cppstd not set

Reply:
Most underlying build files of open source libraries don't follow default standard, they force any standard allowing to properly build if user didn't ask for a specific one.
So if cppstd is not set, I think that it should not be default standard of the compiler, but any standard supported by the compiler and allowing to build the library.

Reply:
Please read the discussion #3977. :-)

Reply:
It's a different issue. Like most libraries, geographiclib doesn't conditionnally disable build of components depending on C++ standard, boost is an edge case.

Reply:
>It's a different issue. 
for the same reason. apple-clang (with c++17 support) was used in C++03 default mode to build the boost.

>conditionnally disable build of components depending on C++ standard, boost is an edge case.
but what about the  conditional compilation depending on C++ standard or other other c++ funny things?

Why not follow the official Conan guide , bump gcc to 6.1 and profit. :-)

Reply:
There is no official Conan guide about cppstd AFAIK, and cppstd in profile is experimental.

Visual Studio don't have C++17 as default standard for example. Do you want to not build pre build binaries at all for Visual Studio if library requires C++17?

Reply:
VS2019 IDE pass the c++14 flag to cl.exe by default.   :-) Xcode 12 does too for apple-clang **but** CCI uses apple-clang in C++03 mode on the command line. 

>Do you want to not provide or test in CCI binaries for Visual Studio if library requires C++17?
Why not, if the underlying build system do right thing ( cmake for example target_compile_features(xxxx _PUBLIC_ cxx_std_17)?  But not by force cppstd in the recipe.

>There is no official Conan guide about cppstd AFAIK, and cppstd in profile is experimental.
IMHO https://docs.conan.io/en/latest/howtos/manage_cpp_standard.html 

No matter simply never use the binaries generated by Conan in the production.

Reply:
> > Do you want to not provide or test in CCI binaries for Visual Studio if library requires C++17?
> 
> Why not, if the underlying build system do right thing ( cmake for example target_compile_features(xxxx _PUBLIC_ cxx_std_17)? But not by force cppstd in the recipe.

Yes and it's what most recipes do, they rely on underlying build system to set any minimum standard, so they don't honor default standard of compiler.